src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -1428,6 +1438,44 @@ bool RTPSParticipantImpl::register_reader(
     return this->mp_builtinProtocols->add_reader(rtps_reader, topic, qos, content_filter);
 }
 
+dds::ReturnCode_t RTPSParticipantImpl::register_reader(
+        RTPSReader* rtps_reader,
+        const TopicDescription& topic,
+        const SubscriptionBuiltinTopicData& sub_builtin_topic_data,
+        const ContentFilterProperty* content_filter)
+{
+    // Check if optional QoS serialization is enabled in the participant properties.
+    // If so, the optional QoS in the builtin topic data needs to be sent.
+    bool should_send_opt_qos = should_send_optional_qos();
+    return this->mp_builtinProtocols->add_reader(rtps_reader, topic, sub_builtin_topic_data, should_send_opt_qos,
+                   content_filter);
+}
+
+bool RTPSParticipantImpl::should_send_optional_qos() const
+{
+    bool should_send_opt_qos = false;
+    if (m_att.properties.properties().size() > 0)
+    {
+        auto serialize_opt_qos_property = std::find_if(
+            m_att.properties.properties().begin(),
+            m_att.properties.properties().end(),
+            [](const fastdds::rtps::Property& property)
+            {
+                return property.name() == fastdds::dds::parameter_serialize_optional_qos;
+            });
+
+        if (serialize_opt_qos_property != m_att.properties.properties().end())
+        {
+            if (serialize_opt_qos_property->value() == "true" ||
+                    serialize_opt_qos_property->value() == "1")
+            {
+                should_send_opt_qos = true;
+            }
+        }
```

Comment:
I think we could simplify this by taking advantage of existing methods, and implement the `PropertyParser::as_bool()` 
```suggestion
        const Property* const serialize_optional_qos_property =
            PropertyPolicyHelper::get_property(m_att.properties, fastdds::dds::parameter_serialize_optional_qos);
        
        if (serialize_optional_qos_property != nullptr)
        {
            should_send_opt_qos = PropertyParser::as_bool(*serialize_optional_qos_property, exception("..."));
        }
```

Reply:
I have added the method `as_bool` to the `ProperyParser`